test/e2e/redirect_port_stripping_test.go

```diff
@@ -0,0 +1,139 @@
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller deployment: %v", err)
+	}
+	service := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.InternalIngressControllerServiceName(ic), service); err != nil {
+		t.Fatalf("failed to get ingresscontroller service: %v", err)
+	}
+
+	echoPod := buildEchoPod(name+"-echo", deployment.Namespace)
+	if err := kclient.Create(context.TODO(), echoPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", echoPod.Namespace, echoPod.Name, err)
+	}
+	defer func() {
+		kclient.Delete(context.TODO(), echoPod)
+	}()
+
+	echoService := buildEchoService(echoPod.Name, echoPod.Namespace, echoPod.ObjectMeta.Labels)
+	if err := kclient.Create(context.TODO(), echoService); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+	}
+	defer func() {
+		kclient.Delete(context.TODO(), echoService)
+	}()
+
+	echoRoute := buildRoute(echoPod.Name, echoPod.Namespace, echoService.Name)
+	echoRoute.Spec.TLS = &routev1.TLSConfig{
+		Termination:                   routev1.TLSTerminationEdge,
+		InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
+	}
+	if err := kclient.Create(context.TODO(), echoRoute); err != nil {
+		t.Fatalf("failed to create route %s/%s: %v", echoRoute.Namespace, echoRoute.Name, err)
+	}
+	defer func() {
+		kclient.Delete(context.TODO(), echoRoute)
+	}()
+
+	// Host header with port 80
+	hostWithPort := echoRoute.Spec.Host + ":80"
+
+	extraCurlArgs := []string{" -i", "-H", "Host: " + hostWithPort}
+
+	clientPodImage := deployment.Spec.Template.Spec.Containers[0].Image
+	clientPod := buildCurlPod(name+"-client", echoRoute.Namespace, clientPodImage, echoRoute.Spec.Host, service.Spec.ClusterIP, extraCurlArgs...)
+
+	if err := kclient.Create(context.TODO(), clientPod); err != nil {
+		t.Fatalf("failed to create client pod: %v", err)
+	}
+	defer func() {
+		kclient.Delete(context.TODO(), clientPod)
+	}()
+
+	err = wait.PollImmediate(1*time.Second, 2*time.Minute, func() (bool, error) {
+		readCloser, err := kubeClient.CoreV1().Pods(clientPod.Namespace).GetLogs(clientPod.Name, &corev1.PodLogOptions{
+			Container: "curl",
+		}).Stream(context.TODO())
+		if err != nil {
+			return false, nil
+		}
+		defer readCloser.Close()
+
+		scanner := bufio.NewScanner(readCloser)
+		for scanner.Scan() {
+			line := scanner.Text()
+			// Looking for "Location: https://<host><path>"
+			// We want to ensure NO ":80" is present.
+			if strings.HasPrefix(strings.ToLower(line), "location:") {
+				t.Logf("Found location header: %s", line)
+				if strings.Contains(line, "https://"+echoRoute.Spec.Host) && !strings.Contains(line, ":80") {
```

Comment:
`redirect sheme` is appending `/` for requests without any path, we may want to make sure `http-request redirect` will be doing the same thing:
```suggestion
				if strings.Contains(line, "https://"+echoRoute.Spec.Host+"/") && !strings.Contains(line, ":80") {
```